gems/aws-sdk-core/lib/aws-sdk-core/cbor/tagged.rb

```diff
@@ -0,0 +1,18 @@
+# frozen_string_literal: true
+
+module Aws
+  module Cbor
+    Tagged = Struct.new(:tag, :value) do
```

Comment:
the `cbor` gem also defines its own Tagged class: https://github.com/cabo/cbor-ruby/blob/master/doclib/cbor/tagged.rb#L6
Which it will return when decoding items.

I think we have two options on how to handle this:
1. In our cbor gem engine adaptor, we replace any Cbor Tagged objects with Aws::Taged objects.
2. We ensure the intrefaces are the same (ie, remove the use of Struct and just define a class with attr accessors) for our Tagged object.  Then just rely no ruby's duck typing.